server/listen.ts

```diff
@@ -0,0 +1,153 @@
+import fs from "fs"
+import dotenv from "dotenv"
+import { http, createPublicClient, Address, parseAbi } from 'viem'
+import pool from "./pg"
+import OFFER_ABI from '../contract/artifacts/contracts/Offer.sol/Offer.json'
+import { getPrice } from "../app/src/helpers/assets"
+
+dotenv.config()
+
+const SAVE_FILE_PATH = '.otc'
+
+const BLOCK_INTERVAL = 3
+
+const publicClient = createPublicClient({
+  cacheTime: 0,
+  transport: http(process.env.RPC_URL),
+})
+
+const timestampCache: Record<number, number> = {}
+
+const getTimestamp = async (blockNumber: number) => {
+  if (timestampCache[blockNumber] === undefined) {
+    const block = await publicClient.getBlock({ blockNumber: BigInt(blockNumber) })
+    timestampCache[blockNumber] = Number(block.timestamp) * 1000
+  }
+
+  return timestampCache[blockNumber]
+}
+
+const listen = async () => {
+  let lastBlockNumber = 0
+
+  if (fs.existsSync(SAVE_FILE_PATH)) {
+    lastBlockNumber = Number(fs.readFileSync(SAVE_FILE_PATH))
```

Comment:
Server can be stateless - use db and avoid fs if possible

Reply:
Can listen run on stateless server? as it runs from setInterval

Reply:
Maybe good for now, some of those servers disable fs access. Can fix later when there is a concrete need. Will use a full VM for now